pkg/cmd/cli/cmd/newapp.go

```diff
@@ -144,6 +144,7 @@ func NewCmdNewApplication(fullName string, f *clientcmd.Factory, out io.Writer)
 	cmd.Flags().BoolVar(&config.InsecureRegistry, "insecure-registry", false, "If true, indicates that the referenced Docker images are on insecure registries and should bypass certificate checking")
 	cmd.Flags().BoolVarP(&config.AsList, "list", "L", false, "List all local templates and image streams that can be used to create.")
 	cmd.Flags().BoolVarP(&config.AsSearch, "search", "S", false, "Search all templates, image streams, and Docker images that match the arguments provided.")
+	cmd.Flags().BoolVar(&config.AllowMissing, "allow-missing", false, "If true, indicates that referenced Docker images that cannot be found locally or in a registry should still be used.")
```

Comment:
`allow-missing` isn't a great name.  I'd prefer to be more descriptive here, especially since this flag isn't something a user is going to find automatically.  Probably should be `allow-missing-images` or `use-images-if-missing`